android-pay/src/main/java/com/stripe/wrap/pay/activity/StripeAndroidPayActivity.java

```diff
@@ -605,6 +632,85 @@ protected void onStripePaymentSourceReturned(
 
     /*------ End Overrides ------*/
 
+    private void getStripeSource(@NonNull final FullWallet fullWallet, @NonNull final Token token) {
+        mHandlerThread = new HandlerThread("Stripe networking");
+        mHandlerThread.start();
+        mStripeUiHandler = new Handler(Looper.getMainLooper()) {
+            private FullWallet wallet = fullWallet;
+            private Token originalToken = token;
+
+            @Override
+            public void handleMessage(Message msg) {
+                super.handleMessage(msg);
+                switch (msg.what) {
+                    case MSG_SOURCE_COMPLETE:
+                        Bundle bundle = msg.getData();
+                        mHandlerThread.quit();
+                        mHandlerThread = null;
+                        if (bundle != null) {
+                            String sourceString = bundle.getString("source", "shiiit");
+                            Source source = Source.fromString(sourceString);
+                            if (source != null && wallet != null) {
+                                logApiCallOnNewThread(source, null);
+                                onStripePaymentSourceReturned(wallet, source);
+                                return;
+                            }
+                        }
+
+                        logApiCallOnNewThread(originalToken, null);
+                        onStripePaymentSourceReturned(wallet, originalToken);
```

Comment:
So it looks like this will always attempt to create a source from the token, and fall back to returning a token if source creation fails. I'm not sure that's the behavior we want – it seems like we should always return one or the other, and make it a setting in `AndroidPayConfiguration`? Don't really have much context on the discussion here – thoughts @bdorfman-stripe ?